brvs-rules/src/main/java/iroha/validation/rules/impl/whitelist/UpdateWhitelistRule.java

```diff
@@ -0,0 +1,130 @@
+    long createdTime = transaction.getPayload().getReducedPayload().getCreatedTime();
+
+    return transaction
+        .getPayload()
+        .getReducedPayload()
+        .getCommandsList()
+        .stream()
+        .filter(Command::hasSetAccountDetail)
+        .map(Command::getSetAccountDetail)
+        .filter(cmd -> cmd.getAccountId().equals(clientId)
+            && (cmd.getKey().equals(WhitelistUtils.ETH_WHITELIST_KEY)
+            || cmd.getKey().equals(WhitelistUtils.BTC_WHITELIST_KEY)))
+        .allMatch(tx -> {
+          try {
+            String whitelistKey = tx.getKey();
+            List<String> clientWhitelist = WhitelistUtils.deserializeClientWhitelist(tx.getValue());
+            logger.info("Client " + clientId + " changed whitelist " + whitelistKey + " to "
+                + clientWhitelist);
+
+            // get old whitelist that was set by BRVS as Pairs(address -> validation_time)
+            Map<String, Long> oldWhitelistValidated = WhitelistUtils.getBRVSWhitelist(
+                brvsAccountId,
+                brvsAccountKeyPair,
+                irohaAPI,
+                clientId,
+                whitelistKey
+            );
+
+            // When whitelist is validated
+            long validationTime = System.currentTimeMillis() / 1000 + validationPeriod;
+            logger.info("ValidationTime: " + validationTime);
+
+            // Prepare new whitelist
+            // remove old addresses that user has removed
+            Map<String, Long> newWhitelistValidated = oldWhitelistValidated.entrySet().stream()
+                .filter(address -> clientWhitelist.contains(address.getKey()))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+            // add new from user whitelist with new validationTime
+            newWhitelistValidated.putAll(
+                clientWhitelist.stream()
+                    .filter(address -> !oldWhitelistValidated.containsKey(address))
+                    .collect(Collectors.toMap(a -> a, t -> validationTime))
+            );
+
+            if (newWhitelistValidated.equals(oldWhitelistValidated)) {
+              logger.info("No changes in whitelist, nothing to update");
+            } else {
+              String jsonNewBrvsWhitelist = WhitelistUtils
+                  .serializeBRVSWhitelist(newWhitelistValidated);
+              logger.info("Send whitelist to Iroha: " + jsonNewBrvsWhitelist);
+
+              irohaAPI.transactionSync(
+                  Transaction.builder(brvsAccountId)
+                      .setCreatedTime(createdTime)
+                      .setAccountDetail(clientId, whitelistKey,
+                          WhitelistUtils.irohaEscape(jsonNewBrvsWhitelist))
+                      .sign(brvsAccountKeyPair)
+                      .build()
+              );
+            }
+          } catch (Exception e) {
+            logger.error("Error during parsing JSON ", e);
```

Comment:
Confusing message. This is not the only operation that may fail in `try` block, as I can see.